ui/vr_webgl.js

```diff
@@ -0,0 +1,611 @@
+    const loadMode = this.player_.getLoadMode();
+    const isMSE = loadMode == shaka.Player.LoadMode.MEDIA_SOURCE;
+    if (!this.video_ || this.video_.readyState < 2 ||
+        (!isMSE && this.video_.playbackRate == 0)) {
+      return;
+    }
+    shaka.ui.Matrix4x4.perspective(this.projectionMatrix_,
+        this.fieldOfView_ * Math.PI / 180, 5 / 3.2, 0.1, 100.0);
+
+    this.gl_.useProgram(this.shaderProgram_);
+
+    this.gl_.clear(this.gl_.COLOR_BUFFER_BIT);
+    this.updateViewPort_();
+
+    if (textureUpdate) {
+      this.gl_.activeTexture(this.gl_.TEXTURE0);
+      this.gl_.bindTexture(this.gl_.TEXTURE_2D, this.texture_);
+      this.gl_.pixelStorei(this.gl_.UNPACK_PREMULTIPLY_ALPHA_WEBGL, 0);
+      this.gl_.texImage2D(this.gl_.TEXTURE_2D, 0, this.gl_.RGBA,
+          this.gl_.RGBA, this.gl_.UNSIGNED_BYTE, this.video_);
+    }
+
+    // Update matrix
+    shaka.ui.Matrix4x4.multiply(this.viewProjectionMatrix_,
+        this.viewMatrix_, this.identityMatrix_);
+    shaka.ui.Matrix4x4.multiply(this.viewProjectionMatrix_,
+        this.projectionMatrix_, this.viewProjectionMatrix_);
+
+    // Plumbing
+    // Vertices
+    this.gl_.bindBuffer(this.gl_.ARRAY_BUFFER, this.verticesBuffer_);
+    goog.asserts.assert(this.vertexPositionAttribute_ != null,
+        'Should have a texture attribute!');
+    this.gl_.vertexAttribPointer(
+        this.vertexPositionAttribute_, 3, this.gl_.FLOAT, false, 0, 0);
+    this.gl_.enableVertexAttribArray(this.vertexPositionAttribute_);
+
+    // UVs
+    this.gl_.bindBuffer(
+        this.gl_.ARRAY_BUFFER, this.verticesTextureCoordBuffer_);
+    goog.asserts.assert(this.textureCoordAttribute_ != null,
+        'Should have a texture attribute!');
+    this.gl_.vertexAttribPointer(
+        this.textureCoordAttribute_, 2, this.gl_.FLOAT, false, 0, 0);
+    this.gl_.enableVertexAttribArray(this.textureCoordAttribute_);
+
+    this.gl_.bindBuffer(
+        this.gl_.ELEMENT_ARRAY_BUFFER, this.verticesIndexBuffer_);
+
+    this.setMatrixUniforms_();
+
+    this.gl_.uniform1i(
+        this.gl_.getUniformLocation(this.shaderProgram_, 'uSampler'), 0);
+
+    if (this.stereoscopicMode_) {
+      this.gl_.viewport(0, 0, this.canvas_.width / 2, this.canvas_.height);
+    }
+
+    // Draw
+    this.gl_.drawElements(this.gl_.TRIANGLES,
+        this.sphere_.indices.length, this.gl_.UNSIGNED_SHORT, 0);
+
+    if (this.stereoscopicMode_) {
+      this.gl_.viewport(this.canvas_.width / 2, 0,
+          this.canvas_.width / 2, this.canvas_.height);
+      this.gl_.drawElements(this.gl_.TRIANGLES,
+          this.sphere_.indices.length, this.gl_.UNSIGNED_SHORT, 0);
+    }
+  }
+
+  /**
+   * @private
+   */
+  setMatrixUniforms_() {
+    const pUniform =
+        this.gl_.getUniformLocation(this.shaderProgram_, 'u_VPMatrix');
+    this.gl_.uniformMatrix4fv(pUniform, false, this.viewProjectionMatrix_);
+  }
+
+  /**
+   * @private
+   */
+  updateViewPort_() {
+    const scale = 1.8;
```

Comment:
You should also make a constant for the far value of 50.5 you use below.

Reply:
You found a bug with test values :)